### ISSUE TITLE
OOM in dataloader resulted in unexpected termination of an epoch

### ISSUE THREAD

**Issue by keunwoochoi** [PyTorchLightning/pytorch-lightning#6351]
Tested on Linux, PL 1.1.3.

So, for some reason my dataloader takes a huge memory and because of that, the child process for the dataloader was terminated.
That didn't lead to the complete termination of the training. It was in the middle of the very first epoch, and seems like the trainer module misunderstood that the first epoch is complete. It also seems like the validation process didn't happen at all.
Then `validation_epoch_end()` method was called, which ended up an error because there was no metric to monitor.

At the end, the error messages were not straightforward. I'm sharing so that somehow one can look into this later.

**Comment by edenlightning** [comment]
Can you share a code we can try to reproduce with? Preferably using[ this colab](https://colab.research.google.com/drive/1HvWVVTK8j2Nj52qU4Q4YCyzOm0_aLQF3?usp=sharing)

also have you tried upgarding to 1.2?

**Comment by keunwoochoi** [comment]
Hi, unfortunately the code is relying a lot on internal source code and infrastructure so I can't share anything real. Just wanted to share that such a thing has happened with PL somewhere :)
